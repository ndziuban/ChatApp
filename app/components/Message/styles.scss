.message {
  max-width: 350px;
  display: flex;
  justify-content: flex-start;
  padding: 5px 10px;
  border-radius: 20px;
  color: black;
  align-self: flex-start;
  border: 1px solid #f1f0f0;
  background: #f1f0f0;
  margin: 2px 0;
  word-break: break-word;

  &.sender {
    border: 1px solid #3578e5;
    color: white;
    background: #3578e5;
    justify-content: flex-end;
    align-self: flex-end;
  }

  &.typing {
    font-style: italic;
    color: #999;
    border: none;
    background: none;
  }
}
