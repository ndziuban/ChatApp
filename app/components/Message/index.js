import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

class Message extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className={`message ${this.props.isTypingMessage ? "typing" : ""} ${this.props.isOwnMessage ? "sender" : ""}`}>
        <span>{this.props.text}</span>
      </div>
    );
  }
};

Message.propTypes = {
  text: PropTypes.string.isRequired,
  isOwnMessage: PropTypes.bool,
  isTypingMessage: PropTypes.bool,
};

export default Message;
