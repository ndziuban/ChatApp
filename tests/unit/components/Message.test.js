import React from 'react';
import Message from '../../../app/components/Message';
import renderer from 'react-test-renderer';

it('Message (standard - sender) renders correctly', () => {
  const tree = renderer
    .create(<Message text="Test Message" isOwnMessage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Message (standard - receiver) renders correctly', () => {
  const tree = renderer
    .create(<Message text="Test Message" isOwnMessage={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Message (sender is typing) renders correctly', () => {
  const tree = renderer
    .create(<Message text="Sender is typing..." isOwnMessage={false} isTypingMessage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
