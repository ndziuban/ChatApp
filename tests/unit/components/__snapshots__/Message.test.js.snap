// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message (sender is typing) renders correctly 1`] = `
<div
  className="message typing "
>
  <span>
    Sender is typing...
  </span>
</div>
`;

exports[`Message (standard - receiver) renders correctly 1`] = `
<div
  className="message  "
>
  <span>
    Test Message
  </span>
</div>
`;

exports[`Message (standard - sender) renders correctly 1`] = `
<div
  className="message  sender"
>
  <span>
    Test Message
  </span>
</div>
`;
